src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/CompletionHandlerTest.cs

```diff
@@ -1266,7 +1266,9 @@ public async Task HandleRequestAsync_ProvisionalCompletion()
                 .Callback<ITextBuffer, string, string, CompletionParams, CancellationToken>((textBuffer, method, clientName, completionParams, ct) =>
                 {
                     Assert.Equal(Methods.TextDocumentCompletionName, method);
-                    Assert.Equal(RazorLSPConstants.RazorCSharpLanguageServerName, clientName);
+                    // We're disabling provisional completions for now because of the impact it has on document syncronization
+                    // Assert.Equal(RazorLSPConstants.RazorCSharpLanguageServerName, clientName);
+                    Assert.Equal(RazorLSPConstants.HtmlLanguageServerName, clientName);
```

Comment:
Would skipping be better? Or given it's a boolean we also have the option of making it non-static and changing it's value here

Reply:
OHH, you're right I should make ProvisionalCompletionEnabled virtaul and test accordingly.